src/config/mod.rs

```diff
@@ -187,6 +189,13 @@ pub trait SinkConfig: core::fmt::Debug + Send + Sync {
     fn input_type(&self) -> DataType;
 
     fn sink_type(&self) -> &'static str;
+
+    /// Resources that the sink is using.
+    /// These resources shouldn't be used in the build method as that can result
+    /// in a build error. Only the sinks are contrained by this.
+    fn resources(&self) -> Vec<Resource> {
```

Comment:
If we make this return a `HashSet` instead of `Vec` we could maybe simplify the `conflicts` method and just verify that the `intersection` with every other set is empty.

Reply:
It would make it a bit simpler, but it would also make impl of `resources` function a bit more verbose. There is also one more reason, I see error reporting of conflicts (Todo: add issue link here) to be better grouped around resources than components, so current version of `conflicts` will be more natural to extend.